crates/compiler/gen_dev/src/generic64/x86_64.rs

```diff
@@ -1415,6 +1415,66 @@ impl Assembler<X86_64GeneralReg, X86_64FloatReg> for X86_64Assembler {
     fn xor_reg64_reg64_reg64(buf: &mut Vec<'_, u8>, dst: Reg64, src1: Reg64, src2: Reg64) {
         binop_move_src_to_dst_reg64(buf, xor_reg64_reg64, dst, src1, src2)
     }
+
+    fn shl_reg64_reg64_reg64<'a, ASM, CC>(
+        buf: &mut Vec<'a, u8>,
+        storage_manager: &mut StorageManager<'a, X86_64GeneralReg, X86_64FloatReg, ASM, CC>,
+        dst: X86_64GeneralReg,
+        src1: X86_64GeneralReg,
+        src2: X86_64GeneralReg,
+    ) where
+        ASM: Assembler<X86_64GeneralReg, X86_64FloatReg>,
+        CC: CallConv<X86_64GeneralReg, X86_64FloatReg, ASM>,
+    {
+        use crate::generic64::RegStorage;
+
+        storage_manager.ensure_reg_free(buf, RegStorage::General(X86_64GeneralReg::RCX));
+
+        mov_reg64_reg64(buf, dst, src1);
```

Comment:
dst or src1 could bet RCX? That would break this code?

Reply:
yes, that is annoying. 

I added some logic for these specific cases. But I'm not entirely sure how this would interact with `storage_manager.ensure_reg_free`. I think it is still needed, but I'm guessing the ordering also matters (otherwise a temporary register might pick RCX again, right?).

So, this gets kind of nasty. Do you have ideas for how to fix this properly?